src/App.tsx

```diff
@@ -14,11 +14,80 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  isVisibleButton: boolean,
+  goodsList: string[],
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    isVisibleButton: true,
```

Comment:
save in state list visibility
```suggestion
    isListVisible: false,
```